test_runner/regress/test_readonly_node.py

```diff
@@ -169,23 +172,63 @@ def generate_updates_on_main(
             )
         return last_flush_lsn
 
-    def trigger_gc_and_select(env: NeonEnv, ep_static: Endpoint, ctx: str):
+    def get_layers_protected_by_lease(
+        ps_http: PageserverHttpClient,
+        tenant_id: Union[TenantId, TenantShardId],
+        timeline_id: TimelineId,
+        lease_lsn: Lsn,
+    ) -> set[str]:
+        """Get all layers whose start_lsn is less than or equal to the lease lsn."""
+        layer_map_info = ps_http.layer_map_info(tenant_id, timeline_id)
+        return set(
+            x.layer_file_name
+            for x in layer_map_info.historic_layers
```

Comment:
should this include open and frozen layers? If a layer is compacted in the meantime, then `issubset` assertion will fail, no?

Reply:
I think we don't need to. First, leases do not guard in-memory layers. We also have background compaction disabled for the test (`"compaction_period": "0s"`).